atlp-rwanda/atlp-devpulse-fn
# Fix/update trainee info

# Fixes Bugs in Update trainee applicant

## Summary

This PR adds validation for trainee applicant form fields and fixes issues related to its submission.

## Related Issue

Fixes #131 

## Changes made

1. Added validation to ensure that all required fields are filled before submitting the form. 
2. Fixed date conversion issues.
3. Updated error messages for better clarity.
4. Dynamic Form Population:
   - Implemented conditional rendering to populate the form 
     fields with the correct values from the current trainee data.
   - Ensured that the form correctly reflects any updates made 
      to the trainee’s information.
5. enhanced the handling of state updates for the trainee data form, ensuring a smoother and more efficient update process for trainee information.

## How to test

On this repo, checkout to branch `fix/update-trainee-info`, then
1.Go to the trainee applicants page -> click on three dots on random trainee -> click on edit -> trainee update form
2. Try submitting the form to check for validation messages
3. Enter invalid data in the form fields (phone, scores, etc) and check for validation messages.
4. Enter valid form data and submit to check for successful message of updating the trainee applicant information. 

## Screenshots


    
![1](https://github.com/user-attachments/assets/8dca304b-e81a-4113-a7b7-2e00bc73592f)

![2](https://github.com/user-attachments/assets/3a41c7af-b9f1-4653-8542-e37a119219c1)

![3](https://github.com/user-attachments/assets/e7303d18-330c-4018-9647-423167a185f9)

![4](https://github.com/user-attachments/assets/644c3b12-9437-47ef-baa0-75f5e6de6652)

### Notes

 Additionally, this PR includes code refactoring to improve readability and maintainability.
